keras/layers/normalization/batch_normalization.py

```diff
@@ -187,6 +197,65 @@ def build(self, input_shape):
 
     def compute_output_shape(self, input_shape):
         return input_shape
+    
+
+    def calculate_mean_and_var(self, inputs):
+
+        if not self.synchronized:
+            return ops.moments(
+                inputs, axes=self._reduction_axes, keepdims=True
+            )
+        else:
+            return self.calulate_sync_mean_and_var(inputs)
+
+
+    def calulate_sync_mean_and_var(self, inputs):
```

Comment:
Since it's only used once you can merge it with the method above (you don't even need the indented `else:` since the first branch returns)